dist/Storable/Storable.xs

```diff
@@ -2960,6 +2960,19 @@ static int store_hash(pTHX_ stcxt_t *cxt, HV *hv)
             keyval = SvPV(key, keylen_tmp);
             keylen = keylen_tmp;
             if (SvUTF8(key)) {
+
+#ifdef utf8_to_bytes_overwrite
+
+                /* If we are able to downgrade here; that means that we have a
+                 * key which only had chars 0-255, but was utf8 encoded.  */
+                if (utf8_to_bytes_overwrite( (U8**) &keyval, &keylen_tmp)) {
```

Comment:
I was a bit worried about this, since this modifies the PV belonging to a SV, but the SV is a copy of the key from the hash on this branch, so it's safe.

It does make a UTF-8 SV with a validly encoded PV into a UTF-8 SV with a probably invalidly encoded PV, but that SV is almost immediately discarded.

Will the new utf8_to_bytes_* functions be added to ppport.h?

Reply:
@khwilliamson could you respond to @tonycoz's concerns above?  Thx.

Reply:
Yes they will